File: flutter_app/lib/features/podcasts/models/podcast_model.dart
class PodcastModel {
  final String id;
  final String title;
  final String description;
  final String coverUrl;
  final String category;
  final int totalEpisodes;
  final String? creatorId;

  PodcastModel({
    required this.id,
    required this.title,
    required this.description,
    required this.coverUrl,
    required this.category,
    required this.totalEpisodes,
    this.creatorId,
  });

  factory PodcastModel.fromJson(Map<String, dynamic> json) {
    return PodcastModel(
      id: json['_id']?.toString() ?? '',
      title: json['title'] ?? '',
      description: json['description'] ?? '',
      coverUrl: json['coverUrl'] ?? '',
      category: json['category'] ?? '',
      totalEpisodes: json['totalEpisodes'] ?? 0,
      creatorId: json['creatorId']?.toString(),
    );
  }
}

File: backend/saran-backend/src/models/podcast.model.js
import mongoose from "mongoose";

const podcastSchema = new mongoose.Schema(
  {
    title: {
      type: String,
      required: true,
      trim: true,
    },

    description: {
      type: String,
      default: "",
    },

    coverUrl: {
      type: String,
      default: "",
    },

    category: {
      type: String,
      default: "general",
    },

    creatorId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      default: null,
    },

    followers: [
      {
        type: mongoose.Schema.Types.ObjectId,
        ref: "User",
      },
    ],

    totalEpisodes: {
      type: Number,
      default: 0,
    },

    isActive: {
      type: Boolean,
      default: true,
    },
  },
  { timestamps: true }
);

export default mongoose.model("Podcast", podcastSchema);